Mathlib/Combinatorics/Additive/Corner/Defs.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Combinatorics.Additive.FreimanHom
+
+/-!
+# Corners
+
+This file defines corners, namely triples of the form `(x, y), (x, y + d), (x + d, y)`, and the
+property of being corner-free the corners theorem and Roth's theorem.
+
+## References
+
+* [Yaël Dillies, Bhavik Mehta, *Formalising Szemerédi’s Regularity Lemma in Lean*][srl_itp]
+* [Wikipedia, *Corners theorem*](https://en.wikipedia.org/wiki/Corners_theorem)
+-/
+
+open Set
+
+variable {G H : Type*}
+
+section AddCommMonoid
+variable [AddCommMonoid G] [AddCommMonoid H] {A B : Set (G × G)} {s : Set G} {t : Set H} {f : G → H}
+  {a b c x₁ y₁ x₂ y₂ : G}
+
+/-- A **corner** of a set `A` in an abelian group is a triple of points of the form
```

Comment:
```suggestion
/-- A **corner** of a set `A` in a commutative monoid is a triple of points of the form
```

There's one more below.

Reply:
I meant that as a "convenient lie", but maybe it is just confusing?

Reply:
Ohh I thought it's just a mistake. I don't think it's confusing, maybe it's okay to leave it as it is. You can decide

Reply:
Actually I maintain my current phrasing because the sentence says `(x, y), (x + d, y), (x, y + d)` (which is the usual definition), and this is the correct definition only in abelian groups.